src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java

```diff
@@ -79,6 +82,25 @@ public ZipFileStatus getStatusFor(String zipFileName) {
         return zipFileStatusList;
     }
 
+    public ZipFileStatus getStatusByCcdId(String ccdId) {
+
+        List<Envelope> envelopes = envelopeRepo.findByCcdId(ccdId);
+        if (envelopes.size() > 0) {
+            String zipFileName = envelopes.get(0).getZipFileName();
+            List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
+
+            return new ZipFileStatus(zipFileName,
+                                     ccdId,
+                                     envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                                     events.stream().map(this::mapEvent).collect(toList()));
```

Comment:
these mappings (for envelopes and events) are duplicated 3 times in this class, this duplication should be eliminated

Reply:
its done